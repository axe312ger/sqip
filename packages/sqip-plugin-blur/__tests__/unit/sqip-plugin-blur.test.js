import SvgPlugin from '../../src/sqip-plugin-blur'

const sampleNoViewBox =
  '<svg><rect fill="#bada55"/><g><path fill="#C0FFEE" d="M51.5 17.5l4 18 15 1z"/></g></svg>'
const sampleNoBg =
  '<svg viewBox="0 0 1024 768"><path fill="#bada55" d="M0 0h1024v640H0z"/><g><path fill="#C0FFEE" d="M51.5 17.5l4 18 15 1z"/></g></svg>'
const sampleWithGroup =
  '<svg viewBox="0 0 1024 768"><rect fill="#bada55"/><g><path fill="#C0FFEE" d="M51.5 17.5l4 18 15 1z"/></g></svg>'
const sampleWithoutGroup =
  '<svg viewBox="0 0 1024 768"><rect fill="#bada55"/><polygon points="0,100 50,25 50,75 100,0" /></svg>'

describe('does prepare svg properly', () => {
  const svgPlugin = new SvgPlugin({})
  test('svg without viewport, not given width & height', () => {
    expect(() =>
      svgPlugin.prepareSVG(sampleNoViewBox, {})
    ).toThrowErrorMatchingSnapshot()
  })
  test('svg without viewport, given width & height', () => {
    const svgPlugin = new SvgPlugin({
      dimensions: {
        width: 1024,
        height: 640
      }
    })
    const result = svgPlugin.prepareSVG(sampleNoViewBox)
    expect(result).toMatchSnapshot()
  })
  test('svg with group, with config', () => {
    const svgPlugin = new SvgPlugin({
      dimensions: {
        width: 1024,
        height: 640
      }
    })
    const result = svgPlugin.prepareSVG(sampleWithGroup)
    expect(result).toMatchSnapshot()
  })
  test('svg without group, config with dimensions only', () => {
    const svgPlugin = new SvgPlugin({
      dimensions: {
        width: 1024,
        height: 640
      }
    })
    const result = svgPlugin.prepareSVG(sampleWithoutGroup)
    expect(result).toMatchSnapshot()
  })
  test('svg with missing background', () => {
    const svgPlugin = new SvgPlugin({
      dimensions: {
        width: 1024,
        height: 640
      }
    })
    expect(() =>
      svgPlugin.prepareSVG(sampleNoBg)
    ).toThrowErrorMatchingSnapshot()
  })
})

describe('applies blur filter', () => {
  test('do nothing when no blur is given', () => {
    const svgPlugin = new SvgPlugin({
      blur: 0
    })
    const result = svgPlugin.applyBlurFilter(sampleWithGroup)
    expect(result).toMatchSnapshot()
  })
  test('svg with group and blur', () => {
    const svgPlugin = new SvgPlugin({
      blur: 5
    })
    const result = svgPlugin.applyBlurFilter(sampleWithGroup)
    expect(result).toMatchSnapshot()
  })
  test('svg without group and blur', () => {
    const svgPlugin = new SvgPlugin({
      blur: 5
    })
    const result = svgPlugin.applyBlurFilter(sampleWithoutGroup)
    expect(result).toMatchSnapshot()
  })
})
