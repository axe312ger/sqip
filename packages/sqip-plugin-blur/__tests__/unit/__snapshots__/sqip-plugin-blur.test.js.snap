// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applies blur filter do nothing when no blur is given 1`] = `"<svg viewBox=\\"0 0 1024 768\\"><rect fill=\\"#bada55\\"/><g><path fill=\\"#C0FFEE\\" d=\\"M51.5 17.5l4 18 15 1z\\"/></g></svg>"`;

exports[`applies blur filter svg with group and blur 1`] = `"<svg viewBox=\\"0 0 1024 768\\"><filter id=\\"b\\"><feGaussianBlur stdDeviation=\\"5\\"/></filter><rect fill=\\"#bada55\\"/><g filter=\\"url(#b)\\"><g><path fill=\\"#C0FFEE\\" d=\\"M51.5 17.5l4 18 15 1z\\"/></g></g></svg>"`;

exports[`applies blur filter svg without group and blur 1`] = `"<svg viewBox=\\"0 0 1024 768\\"><filter id=\\"b\\"><feGaussianBlur stdDeviation=\\"5\\"/></filter><rect fill=\\"#bada55\\"/><g filter=\\"url(#b)\\"><polygon points=\\"0,100 50,25 50,75 100,0\\"/></g></svg>"`;

exports[`does prepare svg properly svg with group, with config 1`] = `"<svg viewBox=\\"0 0 1024 768\\"><rect fill=\\"#bada55\\" width=\\"100%\\" height=\\"100%\\"/><g><path fill=\\"#C0FFEE\\" d=\\"M51.5 17.5l4 18 15 1z\\"/></g></svg>"`;

exports[`does prepare svg properly svg with missing background 1`] = `
"The SVG must have a rect as first shape element which represents the svg background color:

<svg viewBox=\\"0 0 1024 768\\"><path fill=\\"#bada55\\" d=\\"M0 0h1024v640H0z\\"/><g><path fill=\\"#C0FFEE\\" d=\\"M51.5 17.5l4 18 15 1z\\"/></g></svg>"
`;

exports[`does prepare svg properly svg without group, config with dimensions only 1`] = `"<svg viewBox=\\"0 0 1024 768\\"><rect fill=\\"#bada55\\" width=\\"100%\\" height=\\"100%\\"/><polygon points=\\"0,100 50,25 50,75 100,0\\"/></svg>"`;

exports[`does prepare svg properly svg without viewport, given width & height 1`] = `"<svg viewBox=\\"0 0 1024 640\\"><rect fill=\\"#bada55\\" width=\\"100%\\" height=\\"100%\\"/><g><path fill=\\"#C0FFEE\\" d=\\"M51.5 17.5l4 18 15 1z\\"/></g></svg>"`;

exports[`does prepare svg properly svg without viewport, not given width & height 1`] = `
"SVG is missing viewBox attribute while Width and height were not passed:

<svg><rect fill=\\"#bada55\\"/><g><path fill=\\"#C0FFEE\\" d=\\"M51.5 17.5l4 18 15 1z\\"/></g></svg>"
`;
